index.html

```diff
@@ -101,212 +225,1327 @@ <h3>
+
+            while (true) {
+              const { value, done } = await reader.read();
+              if (done) {
+                // |reader| has been canceled.
+                break;
+              }
+              // Do something with |value|...
+            }
+
+            reader.releaseLock();
+          </pre>
+        </aside>
+      
+        The steps to <dfn data-lt="initialize tcp readable stream">initialize the TCP readable stream</dfn> are:
+
+        <ol>
+          <li>
+            Let |stream:ReadableStream| be a [=new=] {{ReadableStream}}.
+          <li>
+            Let |pullAlgorithm| be the following steps:
+            <ol>
+              <li>
+                Run the following steps in parallel:
+              <ol>
+                <li>
+                  Invoke the operating system to read a tcp packet from the socket, placing the result in the [=byte sequence=] |bytes|.
+                <li>
+                  [=Queue a global task=] on the [=relevant global object=] of
+                  [=this=] using the [=TCP socket task source=] to run the
+                  following steps:
+                  <ol>
+                    <li>
+                      If FIN packet was received, run the following steps:
+                      <ol>
+                        <li>
+                          Invoke [=ReadableStream/close=] on [=this=].{{TCPSocket/[[readable]]}}.
+                        <li>
+                          Invoke the steps to [=handle closing the TCP readable stream=].
+                      </ol>
+                    <li>
+                      If no errors were encountered run the following steps:
+                      <ol>
+                        <li>
+                          Let |buffer| be a [=new=] {{ArrayBuffer}} created from |bytes|.
+                        <li>
+                          Let |chunk| be a [=new=] {{Uint8Array}} view over |buffer|,
+                          who's length is the length of |bytes|.
+                        <li>
+                          Invoke [=ReadableStream/enqueue=] on
+                          [=this=].{{TCPSocket/[[readable]]}} with |chunk|.
+                      </ol>
+                    <li>
+                      If an operating system error was encountered, invoke
+                      [=ReadableStream/error=] on [=this=].{{TCPSocket/[[readable]]}} with
+                      a "{{NetworkError}}" {{DOMException}} and invoke the steps
+                      to [=handle closing the TCP readable stream=].
+                </ol>
+              <li>
+                Return [=a promise resolved with=] `undefined`.
+            </ol>
+            
+          <li>
+            Let |cancelAlgorithm| be the following steps:
+            <ol>
+              <li>
+                Let |promise:Promise| be [=a new Promise=].
+              <li>
+                [=Queue a global task=] on the [=relevant global object=] of
```

Comment:
These steps aren't doing anything in parallel anymore so you can simplify this to just invoking "handle closing the TCP readable stream" and returning a resolve promise.

Reply:
Fixed!